contracts/impl/TaskImpl.sol

```diff
@@ -25,18 +25,19 @@ library TaskImpl {
     using Bytes for uint64;
     using Bytes for address;
 
-    event TaskRecordCreated(bytes32 taskId, bytes32 inputsHash, uint64 gasLimit, uint64 gasPx, address sender,
+    event TaskRecordCreated(bytes32 indexed taskId, bytes32 inputsHash, uint64 gasLimit, uint64 gasPx, address sender,
         uint blockNumber);
     // SecretContractDeployed => bytes32s [taskId, preCodeHash, codeHash, initStateDeltaHash]
-    event SecretContractDeployed(uint64 gasUsed, address optionalEthereumContractAddress, bytes32[4] bytes32s,
-        uint gasUsedTotal, bytes optionalEthereumData, address workerAddress);
+    event SecretContractDeployed(bytes32 indexed taskId, uint64 gasUsed, address optionalEthereumContractAddress,
+        bytes32[4] bytes32s, uint gasUsedTotal, bytes optionalEthereumData, address workerAddress);
```

Comment:
Hi @apalepu23, is `scAddr` contained within `bytes32[4] bytes32s`?  
Because we also need to filter on it in here: https://github.com/enigmampc/enigma-p2p/blob/3069a5cdf0d81ed72e9a0307c9c5188a1426072a/src/ethereum/EnigmaContractProductionWriterAPI.js#L135-L139

(Cc: @lenak25, @lacabra)

Reply:
(As @lenak25 mentioned here: https://github.com/enigmampc/enigma-contract/pull/174#pullrequestreview-319702098)

Reply:
hey @assafmo ! the `scAddr` == `taskId`, so the indexed `taskId` field here should address your requirement for `scAddr` as well. let me know if that makes sense!

Reply:
Okay, this helps. Thanks. :-)